File: patch_think.py
import os

path = '/Users/a1412/Desktop/Rcode/Rcode/src/adapters/model/cloudcode-openai-adapter.ts'
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

injection = """  if (input.reasoning_effort && input.reasoning_effort !== 'off') {
    system.push('Before taking action or responding, please think step-by-step and wrap your thinking process strictly inside <think> and </think> tags. Do not skip this step.');
  }"""

content = content.replace(
    "  const toolNameById = new Map<string, string>()",
    injection + "\n\n  const toolNameById = new Map<string, string>()"
)

with open(path, 'w', encoding='utf-8') as f:
    f.write(content)

File: patch_think.js
const fs = require('fs');
const path = '/Users/a1412/Desktop/Rcode/Rcode/src/adapters/model/cloudcode-openai-adapter.ts';
let content = fs.readFileSync(path, 'utf8');

const injection = `
  if (input.reasoning_effort && input.reasoning_effort !== 'off') {
    system.push('Before taking action or responding, please think step-by-step and wrap your thinking process strictly inside <think> and </think> tags.');
  }
`;

content = content.replace(
  "  const toolNameById = new Map<string, string>()",
  injection.trim() + "\n\n  const toolNameById = new Map<string, string>()"
);

fs.writeFileSync(path, content, 'utf8');
